magnum_cluster_api/proxy/utils.py

```diff
@@ -39,9 +40,12 @@ def get_default_ip_address():
     )
     return ip_address
 
-
-def find_free_port():
+def find_free_port(port_hint=0):
     with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
-        s.bind(("", 0))
+        try:
+            s.bind(("", port_hint))
+        except socket.error as e:
+            if e.errno == errno.EADDRINUSE:
+                s.bind(("", 0))
```

Comment:
```suggestion
        except OSError:
            s.bind(("", 0))
```
How about to try a free port if bind failed regardless of the errno?. And i think socket.error was deprecated?

Reply:
Done